ironfish-rust/src/frost_utils/split_spender_key.rs

```diff
@@ -0,0 +1,175 @@
+
+type AuthorizingKey = [u8; 32];
+
+pub struct TrustedDealerKeyPackages {
+    pub verifying_key: AuthorizingKey, // verifying_key is the name given to this field in the frost protocol
+    pub proof_generation_key: ProofGenerationKey,
+    pub view_key: ViewKey,
+    pub incoming_view_key: IncomingViewKey,
+    pub outgoing_view_key: OutgoingViewKey,
+    pub public_address: PublicAddress,
+    pub key_packages: HashMap<Identifier, KeyPackage>,
+    pub public_key_package: PublicKeyPackage,
+}
+
+pub fn split_spender_key(
+    coordinator_sapling_key: SaplingKey,
+    min_signers: u16,
+    max_signers: u16,
+    identifiers: Vec<Identifier>,
+) -> TrustedDealerKeyPackages {
+    let secret = coordinator_sapling_key
+        .spend_authorizing_key
+        .to_bytes()
+        .to_vec();
+
+    let secret_config = SecretShareConfig {
+        min_signers,
+        max_signers,
+        secret,
+    };
+
+    let identifier_list = IdentifierList::Custom(&identifiers);
+
+    let mut rng: rand::prelude::ThreadRng = thread_rng();
+
+    let (key_packages, public_key_package) =
+        split_secret(&secret_config, identifier_list, &mut rng).unwrap();
+
+    let authorizing_key_bytes = public_key_package.verifying_key().serialize();
+
+    let authorizing_key = SubgroupPoint::from_bytes(&authorizing_key_bytes).unwrap();
+
+    let proof_generation_key = ProofGenerationKey {
+        ak: authorizing_key,
+        nsk: coordinator_sapling_key.sapling_proof_generation_key().nsk,
+    };
+
+    let nullifier_deriving_key = *PROOF_GENERATION_KEY_GENERATOR
+        * coordinator_sapling_key.sapling_proof_generation_key().nsk;
+
+    let view_key = ViewKey {
+        authorizing_key,
+        nullifier_deriving_key,
+    };
+
+    let incoming_view_key = coordinator_sapling_key.incoming_view_key().clone();
+
+    let outgoing_view_key: OutgoingViewKey = coordinator_sapling_key.outgoing_view_key().clone();
+
+    let public_address = incoming_view_key.public_address();
+
+    TrustedDealerKeyPackages {
+        verifying_key: authorizing_key_bytes,
+        proof_generation_key,
+        view_key,
+        incoming_view_key,
+        outgoing_view_key,
+        public_address,
+        key_packages,
+        public_key_package,
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use ironfish_frost::{
+        frost::{frost::keys::reconstruct, JubjubBlake2b512},
+        participant::Secret,
+    };
+
+    #[test]
+    fn test_throws_error_with_mismatch_signer_and_identifiers_lengths() {
+        let mut identifiers = Vec::new();
+
+        for _ in 0..10 {
+            identifiers.push(
+                Secret::random(thread_rng())
+                    .to_identity()
+                    .to_frost_identifier(),
+            );
+        }
+
+        let sapling_key = SaplingKey::generate_key();
+
+        let result = std::panic::catch_unwind(|| {
+            split_spender_key(sapling_key, 5, 11, identifiers.clone());
+        });
```

Comment:
the fact that this is how to catch the error makes me think that I should right the function differently with a Result<> return type?